### PR TITLE
Add `options` to all functions; Make it be passed through to dependencies

### PR DESCRIPTION
**Pull request by leshakoss**

### CODE REVIEW

**Comment by LeshaKoss** [comment]
@kossnocorp please review and tell me what to change

File: src/add_hours/index.js [hunk 1]
```diff
@@ -9,15 +9,16 @@ var parse = require('../parse/index.js')
  *
  * @param {Date|String|Number} date - the date to be changed
  * @param {Number} amount - the amount of hours to be added
+ * @param {Object} [options] - the object with options. See [options]{@link docs/types/options}
```
Comment (LeshaKoss) [inline src/add_hours/index.js]: @kossnocorp 
` * @param {Object} [options]`
or
` * @param {Options} [options]`
or
` * @param {options} [options]`
?

File: docs/types/options.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ *   weekStartsOn: 1,
+ *   additionalDigits: 1,
+ *   locale: eoLocale
+ * }
+ * var result = format(startOfWeek('+12345-12-15', options), 'dddd, D MMMM YYYY', options)
+ * //=> 'lundo, 10 decembro 12345'
+ */
+var Options = {}
+
+module.exports = Options
```
Comment (kossnocorp) [inline docs/types/options.js]: Same question as here: https://github.com/date-fns/date-fns/pull/382#discussion_r95149935

File: src/add_days/index.js [hunk 3]
```diff
@@ -9,15 +9,16 @@ var parse = require('../parse/index.js')
  *
  * @param {Date|String|Number} date - the date to be changed
  * @param {Number} amount - the amount of days to be added
+ * @param {Options} [options] - the object with options. See [Options]{@link docs/types/Options}
```
Comment (kossnocorp) [inline src/add_days/index.js]: Here and below: can we mark it as an optional argument?

File: CHANGELOG.md [hunk 4]
```diff
@@ -24,6 +24,8 @@ rather than spread arguments.
   var maxDate = min([new Date(1989, 6 /* Jul */, 10), new Date(1987, 1 /* Feb */, 11)])
   ```
 
+- Every function now has `options` as the last argument which is passed to all its dependencies
+  for consistency and future features.
```
Comment (kossnocorp) [inline CHANGELOG.md]: Please add a link to `docs/Options.js`

Comment (leshakoss) [inline CHANGELOG.md]: Do you mean, like https://date-fns.org/docs/Options ?

Comment (kossnocorp) [inline CHANGELOG.md]: I mean to the file at GitHub. https://date-fns.org/docs/types#Options will be the best option, but then we'll have to build a special doc + page with the types list (which is still required, so please create an issue to date-fns/date-fns.org)

**Pull request merged**